src/offline/cable_driver_init.F90

```diff
@@ -96,9 +110,82 @@ SUBROUTINE cable_driver_init(mpi_grp)
     END IF
 
     ! Open, read and close the namelist file.
-    OPEN(10, FILE=CABLE_NAMELIST, STATUS="OLD", ACTION="READ")
-    READ(10, NML=CABLE)
-    CLOSE(10)
+    OPEN(NEWUNIT=unit, FILE=CABLE_NAMELIST, STATUS="OLD", ACTION="READ")
+    READ(unit, NML=CABLE)
+    CLOSE(unit)
+
+    cable_runtime%offline = .TRUE.
+
+    ! Open, read and close the consistency check file.
+    ! Check triggered by cable_user%consistency_check = .TRUE. in cable.nml
+    IF (mpi_grp%rank == 0 .AND. cable_user%consistency_check) THEN
+      OPEN(NEWUNIT=unit, FILE=filename%trunk_sumbal, STATUS='old', ACTION='READ', IOSTAT=ioerror)
+      IF(ioerror == 0) THEN
+        READ(unit, *) trunk_sumbal  ! written by previous trunk version
+      END IF
+      CLOSE(unit)
+    END IF
+
+    ! Open log file:
+    IF (mpi_grp%rank == 0) THEN
+      OPEN(logn, FILE=filename%log)
+    ELSE IF (cable_user%logworker) THEN
+      WRITE(cRank, FMT='(I4.4)') mpi_grp%rank
+      OPEN(NEWUNIT=logn, FILE="cable_log_"//cRank, STATUS="REPLACE")
+    ELSE
+      OPEN(NEWUNIT=logn, FILE="/dev/null")
+    END IF
+
+    IF (IARGC() > 0 .AND. arg_not_namelist) THEN
+      CALL GETARG(1, filename%met)
+      CALL GETARG(2, casafile%cnpipool)
+    END IF
+
+    ! Initialise flags to output individual variables according to group
+    ! options from the namelist file
+    IF (mpi_grp%rank == 0) THEN
+      CALL set_group_output_values()
+    END IF
+
+    ! TODO(Sean): we should not be setting namelist parameters in the following if
+    ! block - all options are all configurable via the namelist file and is
+    ! unclear that these options are being overwritten. A better approach would be
+    ! to error for bad combinations of namelist parameters.
+    IF (icycle > CASAONLY_ICYCLE_MIN) THEN
+      icycle                     = icycle - CASAONLY_ICYCLE_MIN
+      CASAONLY                   = .TRUE.
+      CABLE_USER%CASA_DUMP_READ  = .TRUE.
+      CABLE_USER%CASA_DUMP_WRITE = .FALSE.
+    ELSE IF (icycle == 0) THEN
+      CABLE_USER%CASA_DUMP_READ  = .FALSE.
+      spincasa                   = .FALSE.
+      CABLE_USER%CALL_POP        = .FALSE.
+    END IF
+
+    ! TODO(Sean): overwriting l_casacnp defeats the purpose of it being a namelist
+    ! option - we should either deprecate the l_casacnp option or not overwrite
+    ! its value.
+    l_casacnp = icycle > 0
+
+    IF (l_casacnp .AND. (icycle == 0 .OR. icycle > 3)) THEN
+       STOP 'icycle must be 1 to 3 when using casaCNP'
+    END IF
+    IF (l_laiFeedbk .OR. l_vcmaxFeedbk) THEN
+      STOP 'casaCNP required to get prognostic LAI or Vcmax'
+    END IF
+    IF (l_vcmaxFeedbk .AND. icycle < 1) THEN
+      STOP 'icycle must be 2 to 3 to get prognostic Vcmax'
+    END IF
+    IF (icycle > 0 .AND. (.NOT. soilparmnew)) THEN
+      STOP 'casaCNP must use new soil parameters'
+    END IF
```

Comment:
I just noticed this code is different between serial and MPI drivers. The current code as is will fail whenever `l_laiFeedbk` or `l_vcmaxFeedbk` are true regardless of whether CASACNP is enabled which does not make sense. Also the condition testing `'icycle must be 2 to 3 to get prognostic Vcmax'` should check for `icycle < 2` instead of `icycle < 1`. The MPI driver implementation looks to be the correct version here:
```Fortran
    IF( l_casacnp  .AND. ( icycle == 0 .OR. icycle > 3 ) )                   &
         STOP 'icycle must be 1 to 3 when using casaCNP'
    IF( ( l_laiFeedbk .OR. l_vcmaxFeedbk ) .AND. ( .NOT. l_casacnp ) )       &
         STOP 'casaCNP required to get prognostic LAI or Vcmax'
    IF( l_vcmaxFeedbk .AND. icycle < 2 )                                     &
         STOP 'icycle must be 2 to 3 to get prognostic Vcmax'
    IF( icycle > 0 .AND. ( .NOT. soilparmnew ) )                             &
         STOP 'casaCNP must use new soil parameters'
```

Reply:
Good catch! There are probably a few more subtle differences like this one in the code. I'll try to keep an eye for those when reviewing changes.